marklogic-unit-test-modules/src/main/ml-modules/root/test/test-helper.xqy

```diff
@@ -258,6 +276,20 @@ declare function helper:assert-equal($expected as item()*, $actual as item()*, $
     fn:error(xs:QName("ASSERT-EQUAL-FAILED"), "Assert Equal failed", ($expected, $actual, " : ", $error-object))
 };
 
+declare function helper:assert-equal-message($expected as item()*, $actual as item()*, $message as xs:string) {
```

Comment:
I'm not a fan of this naming convention. With the existing `assert-equal`, I think you can pass in a string as the error object and get the same result:

`test:assert-equal($foo/a, $foo/b, "these things are different")`

If that's correct, then I favor dropping this function. The function below could be kept, still named `assert-equal-message`, but with the four parameters.